doc/task_breakdown.md

```diff
@@ -0,0 +1,70 @@
+# Task Breakdown for SQL Query Builder Development
+
+## 1. Core Modules of the Program
+
+### 1.1 Table Management Module
+- Tasks:
+  - Implement functionality to display a list of tables from the database.
+  - Add the ability to select multiple tables simultaneously.
+  - Configure the removal of tables from the current selection set.
+- Technical Details:
+  - Use an API to fetch the list of tables from the database.
+  - Ensure that removing a table correctly updates associated data (sorting, filters, relationships).
+
+### 1.2 Relationship Management Module
+- Tasks:
+  - Implement automatic detection of relationships between tables (based on foreign keys).
```

Comment:
again, you are trying to make your life harder, if you just add join condition as property for a table, it will be much simplier